src/liquid/components/ld-input/ld-input.tsx

```diff
@@ -279,19 +294,29 @@ export class LdInput implements InnerFocusable, ClonesAttributes {
     registerAutofocus(this.autofocus)
   }
 
+  private isInputTypeFile = (
+    input: HTMLInputElement | HTMLTextAreaElement
+  ): input is HTMLInputElement => {
+    return (input as HTMLInputElement).type === 'file'
+  }
+
   private handleChange = (ev: InputEvent) => {
     this.el.dispatchEvent(new InputEvent('change', ev))
+
+    if (this.isInputTypeFile(this.input)) {
+      this.ldchangefile.emit(this.input.files)
```

Comment:
I wonder if it'd be the better API to not introduce a seperate custom event for files. What are your arguments agains using onLdchange and pass either value OR files?

Reply:
An alternative approach would be to add a read-only `files` prop and do something like this to maintain backward compatibility with the `CustomEvent<string>` type of the `ldchange` event:

```suggestion
      this.files = this.input.files
      this.ldchange.emit(`${this.input.files.length}`)
```

What do you guys think about this?

Reply:
Just for clarification: returning something other than `string` with the `ldchange` event would result in type errors/warnings for previous uses of the event.

Accessing the files of the `ld-input` component would look something like:

```jsx
<LdChange onLdchange={(event) => {
  if (event.target.files.length) doSomething(event.target.files)
}} />
```

Reply:
I like the idea in general. Though I have implemented it differently now, using a `Object.defineProperty` with a getter. Please have a look. 👀